tools/crm_simulate.c

```diff
@@ -26,11 +26,44 @@
 #include <crm/pengine/status.h>
 #include <pacemaker-internal.h>
 
+/* show_scores and show_utilization can't be added to this struct.  They
+ * actually come from include/pcmki/pcmki_scheduler.h where they are
+ * defined as extern.
+ */
+struct {
+    gboolean all_actions;
+    char *dot_file;
+    char *graph_file;
+    const char *input_file;
```

Comment:
I believe all of these should be char* with dup'ing optarg.

Reply:
As long as it's marked G_OPTION_FILENAME, glib will allocate a new gchar * to store it in, which means I need to do that and then g_free them.